File: JavaScript_Basic_04_2023_SolutionsOfProblems/05.Ex_PB-JS-While-Loop/P01.OldBooks.js
function oldBooks(input) {
    let wantedBook = input[0];
    let i = 1;
    while (input[i] !== "No More Books" && input[i] !== wantedBook) {
        i++;
    }
    if (input[i] === wantedBook) {
        console.log(`You checked ${i - 1} books and found it.`);
    } else {
        console.log("The book you search is not here!");
        console.log(`You checked ${i - 1} books.`);
    }
}
oldBooks(["The Spot",
    "Hunger Games",
    "Harry Potter",
    "Torronto",
    "Spotify",
    "No More Books"])

File: Java_Basic_Jan_2023_SolutionsOfProblems/P05Ex_While_Loop/P01_OldBooks.java
package P05Ex_While_Loop;

import java.util.Scanner;

public class P01_OldBooks {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String nameWantedBook = scanner.nextLine();
        String nameCurrentBook = scanner.nextLine();
        int counter = 0;
        while (!nameCurrentBook.equals(nameWantedBook) && !nameCurrentBook.equals("No More Books")) {
            nameCurrentBook = scanner.nextLine();
            counter++;
        }
        if (nameCurrentBook.equals("No More Books")) {
            System.out.println("The book you search is not here!");
            System.out.printf("You checked %d books.", counter);
        } else {
            System.out.printf("You checked %d books and found it.", counter);
        }
    }
}
